File: src/1608/S1608_v1.java
class Solution {
    public int specialArray(int[] nums) {
        Arrays.sort(nums);
        int r = nums.length;
        if (nums[0] >= r) { return r; }
        for (int i=1; i<r; i++) {
            if (nums[r-i] >= i && (nums[(r-i)-1] < i)) {
                return i;
            }
        }
        return -1;
    }
}

File: src/1608/S1608_v1.rs
impl Solution {
    pub fn special_array(nums: Vec<i32>) -> i32 {
        let mut n = nums;
        n.sort();
        n.reverse();
        let r = n.len();
        if n[r-1] >= r as i32 { return r as i32; }
        for i in 1..r {
            if n[i] < i as i32 && n[i-1] >= i as i32 {
                return i as i32;
            }
        }
        -1
    }
}
